components/elm/src/biogeophys/BareGroundFluxesMod.F90

```diff
@@ -266,7 +314,27 @@ subroutine BareGroundFluxes(bounds, num_nolakeurbanp, filter_nolakeurbanp, &
             obu(p) = zldis(p)/zeta
          end do
 
-      end do ! end stability iteration
+         ! Test for convergence
+         iter_final = iter
+         if (iter >= itmin) then
+            fnold = fn
+            fn = 0
+            do f = 1, fnold
+               p = filterp(f)
+               if (.not. (abs(tau_diff(p)) < dtaumin)) then
+                  fn = fn + 1
+                  filterp(fn) = p
+               end if
+            end do
+            if (fn == 0) then
+               exit ITERATION
+            end if
+         end if
```

Comment:
Is this convergence test active when `implicit_stress=.false.`?

Reply:
Same answer as for the `UrbanFluxesMod` equivalent of this block.